src/frontend/src/views/Upload/components/Metadata/components/Table/components/Row/components/LocationField/index.tsx

```diff
@@ -1,75 +1,119 @@
 }: Props): JSX.Element {
-  const { handleChange, handleBlur, values, touched, errors } = formik;
+  const { handleBlur, setFieldValue, values, touched, errors } = formik;
 
   const errorMessage = touched[fieldKey] && errors[fieldKey];
 
-  const value = values[fieldKey] || "";
+  let value: NamedGisaidLocation | null = null;
+  if (values[fieldKey]) {
+    value = values[fieldKey] as NamedGisaidLocation;
+  }
 
-  return (
-    <StyledTextField
-      select
-      SelectProps={SELECT_PROPS}
-      name={fieldKey}
-      value={value}
-      margin="dense"
-      variant="outlined"
-      onChange={handleChange}
-      onBlur={handleBlur}
-      error={Boolean(errorMessage)}
-      helperText={
-        errorMessage ||
-        (isFirstRow && value && (
-          <ApplyToAllColumn
-            fieldKey={fieldKey}
-            value={value}
-            handleClick={applyToAllColumn}
-          />
-        ))
+  const [state, dispatch] = useReducer(locationSearchReducer, {
+    results: [],
+    searching: false,
+  });
+
+  const searcher = async (query: string): Promise<void> => {
+    if (query.length < 2) {
+      dispatch({ results: [] });
+      return;
+    }
+    dispatch({ searching: true });
+
+    const regex = new RegExp(escapeRegExp(query), "i");
+    const filteredLocationOptions = locations.filter((location) =>
+      regex.test(location.name)
+    );
+    // alphabetical sort
+    // this ensures partial locations (i.e. region, country and divison
+    // but no location) end up on top.
+    const sortedLocationOptions = filteredLocationOptions.sort((a, b) => {
+      if (a.name < b.name) {
+        return -1;
       }
-    >
-      <MenuItem value="" disabled>
-        Select County
-      </MenuItem>
-      {COUNTIES.map((county) => {
-        return (
-          <MenuItem key={county} value={county}>
-            {county}
-            {(county === "California" && (
-              <MenuSubtext>County not specified</MenuSubtext>
-            )) ||
-              null}
-          </MenuItem>
-        );
-      })}
-    </StyledTextField>
-  );
-}
+      if (a.name > b.name) {
+        return 1;
+      }
+      return 0;
+    });
+    dispatch({
+      results: sortedLocationOptions.slice(0, 100),
+      searching: false,
+    });
+  };
+
+  const handleSearchInputChange = (
+    event: React.ChangeEvent<HTMLInputElement>
+  ) => {
+    const query = event?.target?.value ?? "";
+    searcher(query);
+  };
 
-function renderValue(value: unknown): React.ReactNode {
-  return <>{value}</>;
+  const handleLocationChange = (e: DefaultMenuSelectOption | null) => {
+    if (e) {
+      const newLocation = e as NamedGisaidLocation;
+      setFieldValue(fieldKey, newLocation);
+    }
+  };
+
+  return (
+    <StyledDiv onBlur={handleBlur}>
+      <Dropdown
+        label={value?.name || "Search For Location"}
+        value={value}
+        onChange={handleLocationChange}
+        options={state.results}
+        search={true}
```

Comment:
```suggestion
        search
```

in React, `propName={true}` is equivalent to `propName` with no value specified